homeassistant/components/geniushub/climate.py

```diff
@@ -0,0 +1,159 @@
+"""Supports Genius hub to provide climate controls."""
+import asyncio
+import logging
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    STATE_AUTO, STATE_ECO, STATE_HEAT, STATE_MANUAL,
+    SUPPORT_TARGET_TEMPERATURE, SUPPORT_OPERATION_MODE, SUPPORT_ON_OFF)
+from homeassistant.const import (
+    ATTR_TEMPERATURE, TEMP_CELSIUS)
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+GENIUSHUB_SUPPORT_FLAGS = \
+    SUPPORT_TARGET_TEMPERATURE | \
+    SUPPORT_ON_OFF | \
+    SUPPORT_OPERATION_MODE
+
+GENIUSHUB_MAX_TEMP = 28.0
+GENIUSHUB_MIN_TEMP = 4.0
+
+# Genius supports only Off, Override/Boost, Footprint & Timer modes
+HA_OPMODE_TO_GH = {
+    STATE_AUTO: 'timer',
+    STATE_ECO: 'footprint',
+    STATE_MANUAL: 'override',
+}
+GH_OPMODE_OFF = 'off'
+GH_STATE_TO_HA = {
+    'timer': STATE_AUTO,
+    'footprint': STATE_ECO,
+    'away': None,
+    'override': STATE_MANUAL,
+    'early': STATE_HEAT,
+    'test': None,
+    'linked': None,
+    'other': None,
+}  # intentionally missing 'off': None
+GH_DEVICE_STATE_ATTRS = ['temperature', 'type', 'occupied', 'override']
+
+
+async def async_setup_platform(hass, hass_config, async_add_entities,
+                               discovery_info=None):
+    """Set up the Genius hub climate devices."""
+    client = hass.data[DOMAIN]['client']
+
+    zones = []
+    for zone in client.hub.zone_objs:
+        if hasattr(zone, 'temperature'):
+            zones.append(GeniusClimate(client, zone))
+
+    async_add_entities(zones, update_before_add=False)
```

Comment:
False is the default.

Reply:
resolved.

Reply:
@MartinHjelmare I can't mark these as resolved because I am only a contributor, and I didn't submit the original PR...